### ISSUE TITLE
Create NWB Model Interpreter

### ISSUE THREAD

**Issue by tarelli** [openworm/org.geppetto.model.nwb#1]
- [x] Create Model Interpreter skeleton
- [x] Extract a stimulus and a response to test the pipeline
- [x] Plot the stimulus/response in the client
- [x] Extract experiment metadata inside a CompositeType

**Comment by niteshthali08** [comment]
Since NWB file is more than 100MB, we cannot directly push it on GitHub as it has size limit of 100MB. I have used git-lfs https://git-lfs.github.com/ to push this file and which stores the pointer to this file in Github repository while storing file on a remote server.

**Comment by tarelli** [comment]
Some great progress from @niteshthali08 visualising electrophysiology traces from an NWB file! 

<img width="1543" alt="screen shot 2016-04-12 at 17 51 52" src="https://cloud.githubusercontent.com/assets/81127/14580433/51b68518-03c4-11e6-8113-866a4003a2c0.png">

Nitesh, do you want to put in the description of this card the things that are left to be done as we discussed?

**Comment by tarelli** [comment]
@niteshthali08 added some next steps, see the description

**Comment by niteshthali08** [comment]
@tarelli : Extract all sweeps.
I have already written code for this. I have extracted all the sweep numbers and have stored them in a list. Calling readNWBFile() function with this sweep numbers in path will extract data for each of the sweeps.

Experiment metadata also looks similar to what is present in sweeps. Do we need to plot this as well like we did for Sweeps ?

**Comment by tarelli** [comment]
@niteshthali08 for metadata I am referring to whatever information is available regarding the experiment, what traces are we looking at? What protocol? etc. I would expect some structured, descriptive information, text-like, is there anything like that?

Update: these fields, inside \general
![screen shot 2016-04-22 at 01 19 41](https://cloud.githubusercontent.com/assets/81127/14737834/0f753ae4-0876-11e6-9a18-ae3d60e391c0.png)

**Comment by tarelli** [comment]
@niteshthali08 hey Nitesh, how's this coming along?

**Comment by niteshthali08** [comment]
@tarelli : I have written code to read metadata from NWB's general folder. But for some reasons "general" folder from "354190011.nwb" file was inaccessible and hence I wasn't able to read anything from this folder. I wasted lot of time figuring out what is going wrong and then I tried reading it from other file and it worked. So I believe general folder inside "354190011.nwb" have some sort of restrictions may be. 
I am creating a new CompositeType for metadata and that is what is expected right ?

With second NWB file I faced another issue. The type of data inside stimulus and response is of "float" instead of "double" which was expected and hence my code is breaking. I had written code expecting that this data will always be of "float". But now i will have to rewrite this code in more generic way to handle both float and double data type.
